### ISSUE TITLE
Neural Energy Casimir Control. (arXiv:2112.03339v1 [eess.SY])

### ISSUE THREAD

**Issue by CoffeeKumazaki** [CoffeeKumazaki/arXiv#13862]
Neural Energy Casimir Control. (arXiv:2112.03339v1 [eess.SY])<br>
https://ift.tt/3GruaFc<br>
<p>The energy Casimir method is an effective controller design approach to stabilize port-Hamiltonian systems at a desired equilibrium. However, its application relies on finding suitable Casimir and Lyapunov functions, which are generally intractable. In this paper, we propose a neural network-based framework to learn these functions. We show how to achieve equilibrium assignment by adding suitable regularization terms in the optimization cost. We also propose a parameterization of Casimir functions for reducing the training complexity. Moreover, the distance between the equilibrium point of the learned Lyapunov function and the desired equilibrium point is analyzed, which indicates that for small suboptimality gaps, the error decreases linearly with respect to the training loss. Our methods are backed up by simulations on a pendulum system.</p>
<br>
<br>
via eess.SY updates on arXiv.org http://arxiv.org/<br>
